### PR TITLE
[a11y] Add isEnabled semantics flag to  text field

### LINKED ISSUE #143337: [ios] [web] iOS and web Semantics do not support labeling EditTexts as disabled.
### Steps to reproduce

b/322345393

        TextField(
          enabled:false,
          decoration:InputDecoration(
            labelText: 'input label',
          )
      ),

### Expected results

read as 'dimmed' in ios

### Actual results

/

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

### PR DESCRIPTION
**Pull request by hannah-hyj**
Add a semantics flag to   text field to fix https://github.com/flutter/flutter/issues/143337 (in IOS the disabled text field is not read `dimmed`)

internal: b/322345393

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by math1man** [comment]
This is a good change, but it won't fix https://github.com/flutter/flutter/issues/143337. I tried manually wrapping `TextField` in a `Semantics` widget myself and it doesn't solve the issue. There is some bug on the iOS and Web engine side that needs to be fixed.

File: packages/flutter/lib/src/material/text_field.dart [hunk 1]
```diff
@@ -1461,71 +1461,74 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
...
-          autocorrectionTextRectColor: autocorrectionTextRectColor,
-          clipBehavior: widget.clipBehavior,
-          restorationId: 'editable',
-          scribbleEnabled: widget.scribbleEnabled,
-          enableIMEPersonalizedLearning: widget.enableIMEPersonalizedLearning,
-          contentInsertionConfiguration: widget.contentInsertionConfiguration,
-          contextMenuBuilder: widget.contextMenuBuilder,
-          spellCheckConfiguration: spellCheckConfiguration,
-          magnifierConfiguration: widget.magnifierConfiguration ?? TextMagnifier.adaptiveMagnifierConfiguration,
+        child: Semantics(
```
[the remark this answers is not in the document]
Comment (hangyujin) [reply packages/flutter/lib/src/material/text_field.dart:1464]: the Semantics on line 1579 is for maxValueLength and currentValueLength, not for the textfield it self.

[the remark this answers is not in the document]
Comment (math1man) [reply packages/flutter/lib/src/material/text_field.dart:1464]: It doesn't really matter what it's for, does it? It all merges together in the end

**Comment by hangyujin** [comment]
I did some manually IOS testing, if a semantics node has flags [`isTextField` `hasEnabledState` `isReadOnly`], it will read `dimmed` in IOS (as we want). if it has flag [`isTextField` `isReadOnly`], it's not read as  `dimmed`.

 the text field semantics node should have enabled/disabled semantics flag.

File: packages/flutter/lib/src/material/text_field.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (hangyujin) [reply packages/flutter/lib/src/material/text_field.dart:1464]: You're right, i confused it with the text field counter text which is a separated semantics node.

**Review by LongCatIsLooong — APPROVED** [review]

**Comment by hangyujin** [comment]
Reason for revert: broke g3 tests

**Pull request merged**